tests/graph/test_edges.py

```diff
@@ -42,6 +42,35 @@ def test_no_edge(self):
         self.assertRaises(retworkx.NoEdgeBetweenNodes, graph.get_edge_data,
                           node_a, node_b)
 
+    def test_update_edge(self):
+        graph = retworkx.PyGraph()
+        node_a = graph.add_node('a')
+        node_b = graph.add_node('b')
+        graph.add_edge(node_a, node_b, 'not edgy')
+        graph.update_edge(node_a, node_b, 'Edgy')
+        self.assertEqual([(0, 1, 'Edgy')], graph.weighted_edge_list())
+
+    def test_update_edge_no_edge(self):
+        graph = retworkx.PyGraph()
+        node_a = graph.add_node('a')
+        node_b = graph.add_node('b')
+        self.assertRaises(retworkx.NoEdgeBetweenNodes, graph.update_edge,
+                          node_a, node_b, None)
+
+    def test_update_edge_by_index(self):
+        graph = retworkx.PyGraph()
+        node_a = graph.add_node('a')
+        node_b = graph.add_node('b')
+        edge_index = graph.add_edge(node_a, node_b, 'not edgy')
+        graph.update_edge_by_index(edge_index, 'Edgy')
```

Comment:
It may be worth having a test for `by_index` that has multiple edges to choose from? Esp multiple edges between the same nodes

Reply:
Ooh, yep that's a good idea, I added a test in: https://github.com/Qiskit/retworkx/pull/236/commits/de9289040f176942cb220acd806e926e8930a081